npe2/manifest/schema.py

```diff
@@ -334,9 +319,16 @@ def _from_entrypoint(cls, entry_point: EntryPoint) -> PluginManifest:
         fname = match.group("attr")
 
         for loc in spec.submodule_search_locations or []:
-            mf = Path(loc) / fname
-            if mf.exists():
-                return PluginManifest.from_file(mf)
+            mf_file = Path(loc) / fname
+            if mf_file.exists():
+                mf = PluginManifest.from_file(mf_file)
+                if distribution is not None:
+                    meta = PackageMetadata.from_dist_metadata(distribution.metadata)
+                    mf.package_metadata = meta
+
+                    mf.name = mf.name or meta.name  # should we assert?
```

Comment:
assert on what exactly?  

If you're referring to `mf.name==meta.name`, I think we have to think about what to do for the napari builtins case.  I think it's ok not to assert with the caveat that users could run into namespacing issues; future conflicts w pypi packages.

Reply:
Well, this is our chance to assert that the manifest name is the same as the package metadata.  It's a critical point obviously.  So we should either just overwrite it (using package meta), assert equality, or something?

The napari manifest is already changed to "napari", as we discussed

Reply:
> The napari manifest is already changed to "napari", as we discussed

I'd forgotten. It's nice to see actually. I think it works better that way.

Now I prefer adding the assertion that they're the same.  This will avoid all kinds of confusion from typos, makes it easier to track down the plugin for commands, etc... It's better to start more strict.

Reply:
asserted